src/common/errorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+interface ErrorBoundaryProps {
+  errorComponent: React.ReactNode;
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+}
+
+// Error boundaries have to be implemented as class components.
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: { errorComponent: React.ReactNode }) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
```

Comment:
Should we send caught errors to Sentry? 🤔

Reply:
This shouldn't interfere with Sentry (the error still gets caught by Sentry's default handler), just prevents react from tearing down the page. I'll test that this is actually the case.

Reply:
The errors caught by the boundary did infact need to be explicitly sent to Sentry. Fixed.

Reply:
👍